de.variantsync.core/src/main/java/de/variantsync/core/ast/LineBasedParser.java

```diff
@@ -0,0 +1,51 @@
+package de.variantsync.core.ast;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class LineBasedParser {
+
+	public LineBasedParser() {}
+
+	public AST<LineGrammar, String> parseDirectory(Path folder) throws IOException {
+		AST<LineGrammar, String> result;
+		if (Files.isDirectory(folder)) {
+			result = new AST<>(LineGrammar.Directory, folder.getFileName().toString());
+			DirectoryStream<Path> directoryStream = Files.newDirectoryStream(folder);
+			for (Path entry : directoryStream) {
+				result.addChild(parseDirectory(entry));
+			}
+		} else {
+			if (!isBinaryFile(folder)) {
+				result = new AST<>(LineGrammar.TextFile, folder.getFileName().toString());
+				List<String> fileStream = Files.readAllLines(folder);
```

Comment:
Sounds weird because the parameter is called `folder` but in this place we know it is actually a file. Possible fix: Rename `folder` to `path`.